echopype/echodata/combine.py

```diff
@@ -89,6 +90,56 @@ def check_zarr_path(
     return validated_path
 
 
+def _check_channel_selection_form(
+    channel_selection: Optional[Union[List, Dict[str, list]]] = None
+) -> None:
+    """
+    Ensures that the provided user input ``channel_selection`` is in
+    an acceptable form.
+
+    Parameters
+    ----------
+    channel_selection: list of str or dict, optional
+        Specifies what channels should be selected for an ``EchoData`` group
+        with a ``channel`` dimension (before combination).
+    """
+
+    # check that channel selection is None, a list, or a dict
+    if not isinstance(channel_selection, (type(None), list, dict)):
+        raise TypeError("The input channel_selection does not have an acceptable type!")
+
+    if isinstance(channel_selection, list):
+        # make sure each element is a string
+        are_elem_str = [isinstance(elem, str) for elem in channel_selection]
+        if not all(are_elem_str):
+            raise TypeError("Each element of channel_selection must be a string!")
+
+    if isinstance(channel_selection, dict):
+
+        # make sure all keys are strings
+        are_keys_str = [isinstance(elem, str) for elem in channel_selection.keys()]
+        if not all(are_keys_str):
+            raise TypeError("Each key of channel_selection must be a string!")
+
+        # make sure all keys are of the form Sonar/Beam_group
+        are_keys_right_form = ["Sonar/Beam_group" in elem for elem in channel_selection.keys()]
```

Comment:
Seems like we should limit the string to something like`"Sonar/Beam_group(\d{1})"` and use `re.match()`?

Reply:
Sounds great! I have went ahead and implemented your suggestion.